syncer/schema.go

```diff
@@ -74,7 +78,39 @@ func (s *Syncer) OperateSchema(ctx context.Context, req *pb.OperateWorkerSchemaR
 		if err != nil {
 			return "", terror.ErrSchemaTrackerCannotCreateTable.Delegate(err, req.Database, req.Table)
 		}
-		return "", nil
+
+		if !req.Flush && !req.Sync {
+			break
+		}
+
+		ti, err := s.schemaTracker.GetTable(req.Database, req.Table)
+		if err != nil {
+			return "", err
+		}
+
+		if req.Flush {
+			log.L().Info("flush table info", zap.String("table info", newSQL))
+			err = s.checkpoint.FlushPointWithTableInfo(tcontext.NewContext(ctx, log.L()), req.Database, req.Table, ti)
+			if err != nil {
+				return "", err
+			}
+		}
+
+		if req.Sync {
+			if s.cfg.ShardMode != config.ShardOptimistic {
+				log.L().Warn("ignore --sync flag", zap.String("shard mode", s.cfg.ShardMode))
+				break
+			}
+			downSchema, downTable := s.renameShardingSchema(req.Database, req.Table)
+			info := s.optimist.ConstructInfo(req.Database, req.Table, downSchema, downTable, []string{""}, nil, []*model.TableInfo{ti})
```

Comment:
now the length of DDLs will become zero, and fails https://github.com/pingcap/dm/blob/8157e33bd21d85504ba533d7ab5dc1cb26417823/pkg/shardddl/optimism/lock.go#L139-L140

Reply:
Since we expect per ddl per table info. I use `[]string{""}` to construct info for operate-schema. Do you have any idea?

Reply:
oh, I forget that's zero length string, not zero length slice 🤣